ldapauthenticator/ldapauthenticator.py

```diff
@@ -268,11 +295,22 @@ def authenticate(self, handler, data):
         password = data['password']
         # Get LDAP Connection
         def getConnection(userdn, username, password):
-            server = ldap3.Server(
-                self.server_address,
-                port=self.server_port,
-                use_ssl=self.use_ssl
-            )
+            if self.use_ssl:
+                ca_cert = None
+                client_cert = None
+                client_key = None
+                if self.server_ca_file != '':
+                    ca_cert = self.server_ca_file
+                if self.client_certificate_file != '':
+                    client_cert = self.client_certificate_file
+                if self.client_key_file != '':
+                    client_key = self.client_key_file
+                tlsSettings = ldap3.Tls(local_private_key_file=client_key,
+                    local_certificate_file=client_cert,
+                    ca_certs_file=ca_cert, validate=ssl.CERT_REQUIRED)
+                server = ldap3.Server(self.server_address, port=self.server_port, use_ssl=True, tls=tlsSettings)
+            else:
+                server = ldap3.Server(self.server_address, port=self.server_port, use_ssl=False)
```

Comment:
With the last changes it should be now possible to only supply certain options, before that, it was needed to use all the three new settings to work with tls endpoints.